File: php/features.php
<?php
declare(strict_types=1);

// RailwayStationPhotos SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class RailwayStationPhotosFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new RailwayStationPhotosBaseFeature();
            case "test":
                return new RailwayStationPhotosTestFeature();
            default:
                return new RailwayStationPhotosBaseFeature();
        }
    }
}

File: py/features.py
# RailwayStationPhotos SDK feature factory

from feature.base_feature import RailwayStationPhotosBaseFeature
from feature.test_feature import RailwayStationPhotosTestFeature


def _make_feature(name):
    features = {
        "base": lambda: RailwayStationPhotosBaseFeature(),
        "test": lambda: RailwayStationPhotosTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
